packages/go_router/README.md

```diff
@@ -36,7 +36,15 @@ See the API documentation for details on the following topics:
 - [Named routes](https://pub.dev/documentation/go_router/latest/topics/Named%20routes-topic.html)
 - [Error handling](https://pub.dev/documentation/go_router/latest/topics/Error%20handling-topic.html)
 
-## Migration guides
+## Breaking Change
+Only changes that requires manual fix are considered breaking changes. Changes that can be fixed
+through `dart fix` are not considered breaking change in this package.
```

Comment:
This violates a fundamental guideline of the Dart package ecosystem; this seems extremely dangerous. What problem is this change attempting to solve?

Reply:
My original idea is to reduce the incremental speed of major version, as well as number of migration guide that would basically just say run `dart fix`.

Reply:
> this seems extremely dangerous

An in particular, I see only three plausible outcomes from this:
1. Package authors stop using `go_router`. If this is the outcome you want (and I'm not sure why you would want that, but if it is), it should be **extremely clearly** documented. Like, big bold letters at the top of the README saying not to use it in any published package.
2. Package authors pin exact versions of `go_router` to work around this package breaking the basic assumptions of semver, which from an ecosystem collision standpoint is substantially worse than major version bumps.
3. Users of any packages using `go_router` are frequently broken, and both we and package authors see a significant increase in bug reports from confused developers. The former will cost us some added triage time closing issues as "invalid" (and probably arguing with developers who will, correctly IMO, say that it is in fact a bug to undermine semver), and the latter will cost developers time and lead to more and more of 1 and/or 2.

Reply:
I see, so the issue is people will usually pin major version only. In that case, this change probably create more chaos than it intended to solve. Thanks for the input. let me think more a bit, but I will probably revert this proposal

Reply:
> I see, so the issue is people will usually pin major version only.

Yes, the entire Dart package ecosystem, Dart docs about it, and the `dart pub` tool's behaviors, are all predicated on the idea of ranges within a major version as the standard.

> My original idea is to reduce the incremental speed of major version

A cosmetic benefit is definitely not worth undermining basic principles of the package ecosystem, which would create developer pain and confusion.

> Thanks for the input. let me think more a bit, but I will probably revert this proposal

If you decide you still want to do this, please don't proceed without further discussion with me. I have very serious reservations about anything being published under the `flutter.dev` publisher that does not follow the Dart+Flutter team's documented best practices (and the ecosystem's standard practices) for version handling.

Reply:
I have decided to revoke the policy changes, so this will still follow the semver. will definitely let you know if anything comes up.